src/main/java/com/p5/adoption/api/controllers/AnimalController.java

```diff
@@ -16,7 +16,8 @@
     // OLD WAY: NO BOENO @RequestMapping(method = RequestMethod.GET, value = "/hello")
     //call: /api/v1/animals/hello
     @GetMapping("/hello")
-    @PreAuthorize("permitAll()")
+    // aici am problema ca imi merge, chiar daca am comentat "permitAll" si teoretic n-ar trebui deoarece nu este autentificat
+//    @PreAuthorize("permitAll()")
```

Comment:
Stiu ca la curs am security config am adaugat o exceptie pentru /hello si am permis accesul tuturor requesturilor, verifica sa nu ai acolo permis accesul la endpoint-ul asta

Reply:
De asemenea, am observat in postman o chestie.
Daca faci un request cu Auth, el o sa iti salveze un cookie de sesiune si apoi chiar daca selectezi No auth din postman tot va ramane acel cookie de sesiune activ in postman  si te va autentifica in backend. Incearca sa verifici daca e prezent acel cookie in REquest Headers in postman si daca e sa il stergi pentru un request cu No Auth

Reply:
Inca un lucru aici, noi am exemplificat la curs toate modalitatile prin care poti sa faci autentificate si autorizare.

Prima optiune:  http.authorizeRequests() din security config
A doua optiune: prin method security si acele adnotari PreAuthorize

Toate lucrurile in spring merg prin filtre, adica un numar de clase care se executa intr-o ordine.

 Sincer nu stiu exact care e ordinea lor, dar pariez pe faptul ca prima optiune se verifica inainte de a doua optiune. Deci chiar daca avem un @PreAuthorize("permitAll") pe o metoda s-ar putea sa nu putem ajunge acolo fiindca crapa  de la `.anyRequest().authenticated()`, si ar fi nevoie sa adaugam path-ul si in security config cu permitAll().